app/web/db/crud.py

```diff
@@ -23,46 +34,92 @@
+    # NOTE: load_only is for optimization and not obfuscation, use
+    # .with_entities() if needed
+    return (
+        db.query(models.Archive)
+        .filter(not_(models.Archive.deleted))
+        .options(
+            load_only(
+                models.Archive.id,
+                models.Archive.created_at,
+                models.Archive.url,
+                models.Archive.result,
+                models.Archive.store_until,
+            )
+        )
+    )
 
 
-def search_archives_by_url(db: Session, url: str, email: str, read_groups: bool | set[str], read_public: bool, skip: int = 0, limit: int = 100, archived_after: datetime = None, archived_before: datetime = None, absolute_search: bool = False) -> list[models.Archive]:
-    # searches for partial URLs, if email is * no ownership (or read/read_public) filtering happens
+def search_archives_by_url(
+    db: Session,
+    url: str,
+    email: str,
+    read_groups: bool | set[str],
+    read_public: bool,
+    skip: int = 0,
+    limit: int = 100,
+    archived_after: datetime = None,
+    archived_before: datetime = None,
+    absolute_search: bool = False,
+) -> list[models.Archive]:
+    # searches for partial URLs, if email is * no ownership
+    # (or read/read_public) filtering happens
     query = base_query(db)
     if email != ALLOW_ANY_EMAIL:
         or_filters = [models.Archive.author_id == email]
         if read_public:
-            or_filters.append(models.Archive.public == True)
-        if read_groups == True:
+            or_filters.append(models.Archive.public.is_(true()))
+        if read_groups is True:
             or_filters.append(models.Archive.group_id.isnot(None))
         else:
             or_filters.append(models.Archive.group_id.in_(read_groups))
         query = query.filter(or_(*or_filters))
     if absolute_search:
         query = query.filter(models.Archive.url == url)
     else:
-        query = query.filter(models.Archive.url.like(f'%{url}%'))
+        query = query.filter(models.Archive.url.like(f"%{url}%"))
     if archived_after:
         query = query.filter(models.Archive.created_at > archived_after)
     if archived_before:
         query = query.filter(models.Archive.created_at < archived_before)
-    return query.order_by(models.Archive.created_at.desc()).offset(skip).limit(get_limit(limit)).all()
+    return (
+        query.order_by(models.Archive.created_at.desc())
+        .offset(skip)
+        .limit(get_limit(limit))
+        .all()
+    )
 
 
-def search_archives_by_email(db: Session, email: str, skip: int = 0, limit: int = 100):
-    return base_query(db).filter(models.Archive.author_id == email).order_by(models.Archive.created_at.desc()).offset(skip).limit(get_limit(limit)).all()
+def search_archives_by_email(
+    db: Session, email: str, skip: int = 0, limit: int = 100
+):
+    return (
+        base_query(db)
+        .filter(models.Archive.author_id == email)
```

Comment:
Should these double = be single =? There's another line where we use single equals which I think makes more syntactic sense.